app/src/hooks/db/useUserApplications.tsx

```diff
@@ -1,21 +1,29 @@
-import { useQuery } from "@tanstack/react-query"
+import { useQuery, useQueryClient } from "@tanstack/react-query"
 import { useSession } from "next-auth/react"
 
 import { getApplications } from "@/lib/actions/projects"
 import { ApplicationWithDetails } from "@/lib/types"
+import { useEffect, useState } from "react"
 
 export function useUserApplications(): {
   data: ApplicationWithDetails[] | undefined
   isLoading: boolean
-  error: Error | null
 } {
   const { data: session } = useSession()
 
+  const queryClient = useQueryClient()
+
+  useEffect(() => {
+    queryClient.invalidateQueries({
+      queryKey: ["userApplications", session?.user?.id],
+    })
+  }, [session?.user.id])
```

Comment:
What's the reason for invalidating query cache within the components lifecycle? This can produce unwanted side-effects